RFCs/006-DynamicData-Attribute.md

```diff
@@ -0,0 +1,66 @@
+# RFC 006- DynamicData Attribute for Data Driven Tests
+
+## Summary
+This details the MSTest V2 framework attribute "DynamicData" for data driven tests where test data, declared as properties or methods, can be shared between more than one test cases.
+
+## Motivation
+Often times, the inline test data need to be shared between tests. There could be more than one test cases in different files which can use same test data. In most of these situations, user end up declaring same inline data for multiple test cases using "DataRow" attribute. Test framework should provide feature so that test data, declared as property, can be
+* Easily re-used
+* Easily organized
+
+## Detailed Design
+
+### Requirements
+1. Inline test data can be reused by multiple test cases.
+
+### Proposed solution
+Here is a solution that meets the above requirements:
+
+The property storing test data should be declared as static.
+```
+    static IEnumerable<object[]> ReusableTestDataProperty =
+        {
+            get
+            {
+                return new[]
+                    {   new object[] {1, 2, 3},
+                        new object[] {4, 5, 6}
+                    };
+            }
+        }
+
+    static IEnumerable<object[]> ReusableTestDataMethod()
+    {
+        return new[]
+                    {   new object[] {1, 2, 3},
+                        new object[] {4, 5, 6}
+                    }; 
+    }
+```
+
+This property can then be used to associate test data with data driven test case like below:
+```
+    [TestMethod]
+    [DynamicData("ReusableTestDataProperty")]
+    public void DynamicDataTestMethod1(int a, int b, int c)
+    {
+        Assert.IsTrue(1, a%3);
+        Assert.IsTrue(2, b%3);
+        Assert.IsTrue(3, c%3);
+    }
+
+    [TestMethod]
+    [DynamicData("ReusableTestDataMethod")]
+    public void DynamicDataTestMethod2(int a, int b, int c)
```

Comment:
For type mismatch does it make sense to try and course the value using Converter.ChangeType or hard fail?

What happens if you don't supply values for parameters with defaults?